Remove and refactor reference to result column

emyl3 opened the pull request:
# FRONTEND PULL REQUEST

## Related Issue
#4222 

## Changes Proposed
- Remove the reference to the `result` column in the front end so that we can remove references to it in the back end

## Additional Information
N/A

## Testing
1. Go to the "Results" tab of an org
2. Ensure multiplex and covid results are displaying as intended
 
## Screenshots / Demos
<img width="1345" alt="Screen Shot 2022-09-14 at 8 15 28 PM" src="https://user-images.githubusercontent.com/20211771/190290515-b657907e-a104-497b-80e7-78975fdc70d3.png">

## Checklist for Author and Reviewer

### Design
- [ ] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [ ] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### Content
- [ ] Any content changes have been approved by content team

### Support
- [ ] Any changes that might generate new support requests have been flagged to the support team
- [ ] Any changes to support infrastructure have been demo'd to support team

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

### Documentation
- [ ] Any changes to the startup configuration have been documented in the README

emyl3 commented on frontend/src/app/testResults/TestResultsList.test.tsx:
```diff
@@ -664,24 +664,6 @@ describe("TestResultsList", () => {
       expect((await screen.findAllByText("Flu A"))[0]).toBeInTheDocument();
     });
-
-    it("should default to column value and show N/A for Flu results for rows with no multiplex results", async () => {
```
⚠️ removed this test since it is no longer applicable

georgehager replied:
I'm confused about this, it does still show N/A for rows with no multiplex results. Maybe I'm misunderstanding the test, but don't we want to still ensure it renders this way?

georgehager replied:
Or is it that we just want to test it in a different location? e.g. the ResultsTable.test.tsx?

georgehager replied:
Actually nvm, I see it was moved to the other test so with the restructuring it makes sense! don't mind me lol

emmastephenson reviewed (approved):
LGTM, thanks Elisa!

georgehager reviewed (approved):
Looks great! Thanks for all the clean up, I think the code is much easier to read now 😀

emyl3 commented:
Found another instance of `result` being used -- 😮‍💨 sorry everyone 😭

emyl3 commented:
Ready for re-review! Thank you for your patience on this!

emmastephenson commented on frontend/src/app/testResults/mocks/resultsMultiplex.mock.tsx:
```diff
@@ -96,7 +96,7 @@ const TEST_RESULTS_MULTIPLEX = [
     internalId: "7c768a5d-ef90-44cd-8050-b96dd7aaa1d5",
     dateTested: "2021-03-19T19:27:21.052Z",
     result: "POSITIVE",
```
We'll need to take these `result` mocks out once the graphQL backend is updated, right?

emyl3 replied:
Yes, thank you for the reminder! 😸 I was planning on doing that along with these changes https://github.com/CDCgov/prime-simplereport/pull/4259

emmastephenson reviewed (approved):
LGTM, tested locally and everything looks good! No worries about the additional review Elisa, happens to everyone at some point :)

fzhao99 commented:
@emyl3 just for my sanity / awareness, how did you check for instances of results being used? I did a general command-f and combed through the relevant files and the instances I saw that were using results from GraphQL used the new `r.results` shape, but just wanted to double check to make sure there wasn't a better way to do so.

emyl3 commented:
> @emyl3 just for my sanity / awareness, how did you check for instances of results being used? I did a general command-f and combed through the relevant files and the instances I saw that were using results from GraphQL used the new `r.results` shape, but just wanted to double check to make sure there wasn't a better way to do so.
> 
> Also flagging once the backend gets changed, but it looks like a few lines in `TestQueue.tsx` might be able to be deleted with this refactor (looking specifically at 247 and potentially the other references to `result` in that file that I think are just there to maintain backwards compatiability.)

That is a great question, @fzhao99! If you could not tell, I was slowly losing my sanity with this PR so thank you for asking. I started out with the general search of the codebase. Clearly that led to references being missed 😅

What I did after was merge this branch and this https://github.com/CDCgov/prime-simplereport/pull/4259 branch, cleaned up a couple frontend GraphQL calls and then started running the tests and smoke testing the app. Saw it blew up in a couple places with these backend changes in place and fixed them up. And clearly that's not good enough since you found another reference to them hahah...

Thank you, Bob! I'll clean that up in this PR. Let me know if anyone else has a better way to check for references to the `result` column 🥲 🕵️

emmastephenson commented on frontend/src/app/testResults/TestResultDetailsModal.tsx:
```diff
@@ -104,21 +109,22 @@ export const DetachedTestResultDetailsModal = ({ data, closeModal }: Props) => {
+      ? getResultByDiseaseName(results, MULTIPLEX_DISEASES.COVID_19)
+      : "UNKNOWN",
   };
   const multiplexFeatureFlagEnabled = useFeature("multiplexEnabled");
```
Discussed offline - @BobanL, when you write the ticket if you could link it here that'd be great!

BobanL replied:
#4301 was created for this!

emyl3 replied:
Thank you both!

georgehager commented:
> > @emyl3 just for my sanity / awareness, how did you check for instances of results being used? I did a general command-f and combed through the relevant files and the instances I saw that were using results from GraphQL used the new `r.results` shape, but just wanted to double check to make sure there wasn't a better way to do so.
> > Also flagging once the backend gets changed, but it looks like a few lines in `TestQueue.tsx` might be able to be deleted with this refactor (looking specifically at 247 and potentially the other references to `result` in that file that I think are just there to maintain backwards compatiability.)
> 
> That is a great question, @fzhao99! If you could not tell, I was slowly losing my sanity with this PR so thank you for asking. I started out with the general search of the codebase. Clearly that led to references being missed 😅
> 
> What I did after was merge this branch and this #4259 branch, cleaned up a couple frontend GraphQL calls and then started running the tests and smoke testing the app. Saw it blew up in a couple places with these backend changes in place and fixed them up. And clearly that's not good enough since you found another reference to them hahah...
> 
> Thank you, Bob! I'll clean that up in this PR. Let me know if anyone else has a better way to check for references to the `result` column 🥲 🕵️

@emyl3 I think testing it with the backend branch changes is a great idea, and will ultimately catch any issues better than a manual search alone. these refactors are so tricky, I think you've done an awesome job!
